com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/StrongReferencerCacheAdapter.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+package com.avaloq.tools.ddk.xtext.resource;
+
+import org.eclipse.emf.common.notify.Notifier;
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+
+import com.avaloq.tools.ddk.caching.CacheManager;
+import com.avaloq.tools.ddk.caching.MapCache;
+
+
+/**
+ * A generic resource set cache adapter which uses strong references.
+ *
+ * @param <V>
+ *          the type of the stored values
+ */
+public class StrongReferencerCacheAdapter<V> extends AdapterImpl implements IResourceSetCache<V> {

```

Comment:
as far as i can tell the CacheAdapter class is pretty much the same - would it perhaps make sense to add some constructor parameter which would then switch to strong references? (we can use soft ones by default)

Reply:
I will take a look tomorrow at this.